src/components/Main/Results/Utils.ts

```diff
@@ -14,3 +16,40 @@ export function verifyTuple(x: [maybeNumber, maybeNumber]): [number, number] | u
 
   return undefined
 }
+
+export function computeNewEmpiricalCases(
+  timeWindow: number,
+  cumulativeCounts?: CaseCountsDatum[],
+): [maybeNumber[], number] {
+  const newEmpiricalCases: maybeNumber[] = []
+  const deltaDay = Math.floor(timeWindow)
+  const deltaInt = timeWindow - deltaDay
+
+  if (!cumulativeCounts) {
+    return [newEmpiricalCases, deltaDay]
+  }
+
+  cumulativeCounts.forEach((_0, day) => {
+    if (day < deltaDay) {
+      newEmpiricalCases[day] = undefined
+      return
+    }
+
+    const startDay = day - deltaDay
+    const startDayPlus = day - deltaDay - 1
+
+    const nowCases = cumulativeCounts[day].cases
+    const oldCases = cumulativeCounts[startDay].cases
+    const olderCases = cumulativeCounts[startDayPlus]?.cases
+    if (oldCases && nowCases) {
+      const newCases = verifyPositive(
+        olderCases ? (1 - deltaInt) * (nowCases - oldCases) + deltaInt * (nowCases - olderCases) : nowCases - oldCases,
+      )
+      newEmpiricalCases.push(newCases)
+      return
+    }
+    newEmpiricalCases[day] = undefined
+  })
+
+  return [newEmpiricalCases, deltaDay]
+}
```

Comment:
This code is moved mostly unchanged from `DeterministicLinePlot.tsx`. The only difference is that the `verifyPositive` function no longer needs to be passed as a parameter to this function.